impeller/entity/entity_pass.cc

```diff
@@ -174,146 +178,212 @@ bool EntityPass::Render(ContentContext& renderer,
+          element_entity.GetTransformation());
     }
+  }
 
-    if (!command_buffer->SubmitCommands()) {
-      return false;
-    }
+  //--------------------------------------------------------------------------
+  /// Setup subpass element.
+  ///
 
-    return true;
-  };
+  else if (const auto& subpass_ptr =
+               std::get_if<std::unique_ptr<EntityPass>>(&element)) {
+    auto subpass = subpass_ptr->get();
 
-  for (const auto& element : elements_) {
-    Entity element_entity;
-
-    // =========================================================================
-    // Setup entity element for rendering ======================================
-    // =========================================================================
-    if (const auto& entity = std::get_if<Entity>(&element)) {
-      element_entity = *entity;
-      if (!position.IsZero()) {
-        // If the pass image is going to be rendered with a non-zero position,
-        // apply the negative translation to entity copies before rendering them
-        // so that they'll end up rendering to the correct on-screen position.
-        element_entity.SetTransformation(
-            Matrix::MakeTranslation(Vector3(-position)) *
-            element_entity.GetTransformation());
-      }
+    if (subpass->delegate_->CanElide()) {
+      return EntityPass::EntityResult::Skip();
     }
 
-    // =========================================================================
-    // Setup subpass element for rendering =====================================
-    // =========================================================================
-    else if (const auto& subpass_ptr =
-                 std::get_if<std::unique_ptr<EntityPass>>(&element)) {
-      auto subpass = subpass_ptr->get();
-
-      if (subpass->delegate_->CanElide()) {
-        continue;
+    if (subpass->delegate_->CanCollapseIntoParentPass() &&
+        !subpass->backdrop_filter_proc_.has_value()) {
+      // Directly render into the parent target and move on.
+      if (!subpass->OnRender(renderer, pass_context.GetRenderTarget(), position,
+                             position, stencil_depth_floor)) {
+        return EntityPass::EntityResult::Failure();
       }
+      return EntityPass::EntityResult::Skip();
+    }
 
-      if (subpass->delegate_->CanCollapseIntoParentPass()) {
-        // Directly render into the parent target and move on.
-        if (!subpass->RenderInternal(renderer, render_target, position,
-                                     pass_depth, stencil_depth_floor)) {
-          return false;
-        }
-        continue;
-      }
+    std::optional<Rect> subpass_coverage;
 
-      const auto subpass_coverage = GetSubpassCoverage(*subpass);
+    if (subpass->elements_.empty() && backdrop_filter_proc_.has_value()) {
+      // An empty subpass with a backdrop filter should render the backdrop over
+      // the entire coverage of the current pass.
```

Comment:
Actually, even if the subpass is not empty, the presence of a backdrop filter indicates that the entire background will be filtered (output restricted by any existing clips).

Reply:
Hmm, I see... what about when SaveLayer bounds are given? That should still clip the filtered background, right?

Reply:
I think this is fixed now. Added a video of the test demonstrating the clipping behavior in the PR description.

Reply:
Actually no, the saveLayer bounds do not clip the filtered background. I don't even know if they clip the children (subpass) either in this case, Skia fiddle would show the expected behavior.

Reply:
I find this behavior a bit odd, but confirmed in Skia fiddle: https://fiddle.skia.org/c/51c183699c942925d0b13d967f932010

Fixed and updated the demo video in the description.